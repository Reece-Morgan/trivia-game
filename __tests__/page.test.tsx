import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Page from "../app/page";

describe("Page", () => {
  it('Page component renders correctly', () => {
    const tree = renderer
      .create(<Page />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("renders a heading", () => {
    render(<Page />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeInTheDocument();
  });
});
