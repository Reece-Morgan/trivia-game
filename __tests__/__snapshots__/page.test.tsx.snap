// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Page component renders correctly 1`] = `
<main
  className="main"
>
  <div
    className="description"
  >
    <h1>
      Get started by editing 
      <code
        className="code"
      >
        app/page.tsx
      </code>
    </h1>
    <div>
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        By
         
        <img
          alt="Vercel Logo"
          className="vercelLogo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height={24}
          onError={[Function]}
          onLoad={[Function]}
          src="/vercel.svg"
          style={
            {
              "color": "transparent",
            }
          }
          width={100}
        />
      </a>
    </div>
  </div>
  <div
    className="center"
  >
    <img
      alt="Next.js Logo"
      className="logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={37}
      onError={[Function]}
      onLoad={[Function]}
      src="/next.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={180}
    />
  </div>
  <div
    className="grid"
  >
    <a
      className="card"
      href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Docs 
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Find in-depth information about Next.js features and API.
      </p>
    </a>
    <a
      className="card"
      href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Learn 
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Learn about Next.js in an interactive course with quizzes!
      </p>
    </a>
    <a
      className="card"
      href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Templates 
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Explore starter templates for Next.js.
      </p>
    </a>
    <a
      className="card"
      href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Deploy 
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Instantly deploy your Next.js site to a shareable URL with Vercel.
      </p>
    </a>
  </div>
</main>
`;
